app/controller/sdl/RController.js

```diff
@@ -537,26 +498,36 @@ SDL.RController = SDL.SDLController.extend(
       }else if (request.params.moduleType == 'HMI_SETTINGS') {
         module = 'HMI settings';
       }
-
-      var popUp = SDL.PopUp.create().appendTo('body').popupActivate(
-        'Would you like to grant access for ' + appName +
-        ' application for module ' + module + '?',
-        function(result) {
-          FFW.RC.GetInteriorVehicleDataConsentResponse(request, result);
-        }
-      );
-
-      setTimeout(
-        function() {
-          if (popUp && popUp.active) {
-            popUp.deactivate();
-            FFW.RC.sendError(
-              SDL.SDLModel.data.resultCode['TIMED_OUT'], request.id,
-              request.method, 'The resource is in use and the driver did not respond in time'
-            );
+      var moduleIds = request.params.moduleIds;
+      var allowed = [];
+      var timedOutSended = false;
+      moduleIds.forEach(element => {
```

Comment:
Right now when SDL sends a `RC.GetInteriorVehicleDataConsent` request with multiple moduleIds. The popups overlap and the responses are stored in reverse order.

```suggestion
      moduleIds.reverse().forEach(element => {
```

Reply:
@ShobhitAd agree, accepted in https://github.com/smartdevicelink/sdl_hmi/pull/216/commits/336b94e0c126ba5c6ba939991705e000a5fb793b